### PR TITLE
Internal: check distro with ssh commands, instead of image-family name

### PR DESCRIPTION
**Pull request by Subbarker**
## Description

This removes some "magic" from our tests

## Related issue

b/337016736

## Checklist:
- Unit tests
  - [x] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [ ] Integration tests do not apply.
  - [x] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

File: integration_test/agents/agents.go [hunk 1]
```diff
@@ -336,30 +336,39 @@ func CheckServicesNotRunning(ctx context.Context, logger *log.Logger, vm *gce.VM
...
-	} else if gce.IsSUSE(vm.Platform) {
-		cmd = fmt.Sprintf("sudo zypper --non-interactive install %s", pkgsString)
-	} else if strings.HasPrefix(vm.Platform, "debian-") ||
-		strings.HasPrefix(vm.Platform, "ubuntu-") {
-		cmd = fmt.Sprintf("sudo apt-get update; sudo apt-get -y install %s", pkgsString)
-	} else {
-		return fmt.Errorf("tryInstallPackages() doesn't support platform %q", vm.Platform)
+
+	switch vm.OS.ID {
+	case "centos", "rocky":
```
Comment (martijnvans) [inline integration_test/agents/agents.go:345]: what about rhel? what about opensuse?

File: integration_test/gce/gce_testing.go [hunk 2]
```diff
@@ -1106,6 +1113,30 @@ func parseImageSpec(options *VMOptions) error {
...
+			ID: "windows",
+		}, nil
+	}
+	id_output, err := getReleaseInfo(ctx, logger, vm, "ID")
+	if err != nil {
+		return nil, err
+	}
+
+	return &OS{
+		ID: id_output.Stdout,
```
Comment (martijnvans) [inline integration_test/gce/gce_testing.go:1136]: probably want to trim any whitespace from this command output

File: integration_test/gce/gce_testing.go [hunk 3]
```diff
@@ -1262,7 +1293,13 @@ func attemptCreateInstance(ctx context.Context, logger *log.Logger, options VMOp
...
 	}
 
-	if IsSUSE(vm.Platform) {
+	if os, err := getOS(ctx, logger, vm); err != nil {
+		return nil, err
+	} else {
+		vm.OS = *os
+	}
+
+	if vm.OS.ID == "sles" {
```
Comment (martijnvans) [inline integration_test/gce/gce_testing.go:1302]: what about opensuse?

File: kokoro/scripts/test/go_test.sh [hunk 4]
```diff
@@ -111,6 +111,10 @@ function set_platforms() {
   local platforms
   platforms=$(yaml project.yaml "['targets']['${TARGET}']['architectures']['${ARCH}']['test_distros']['representative']")
   # If not a presubmit job, add the exhaustive list of test distros.
+  if [[ "${TEST_SUITE_NAME}" == "ops_agent_test" ]]; then
```
Comment (martijnvans) [inline kokoro/scripts/test/go_test.sh:114]: remember to revert this

File: integration_test/gce/gce_testing.go (hunk 3 above)
Comment (Subbarker) [reply integration_test/gce/gce_testing.go:1302]: it's ID is 'sles'

File: integration_test/gce/gce_testing.go (hunk 2 above)
Comment (Subbarker) [reply integration_test/gce/gce_testing.go:1136]: I ran into that earlier, and fixed it by adding `-n` to the `echo`

File: kokoro/scripts/test/go_test.sh (hunk 4 above)
Comment (Subbarker) [reply kokoro/scripts/test/go_test.sh:114]: aye

File: integration_test/agents/agents.go (hunk 1 above)
Comment (Subbarker) [reply integration_test/agents/agents.go:345]: Fixed

File: integration_test/gce/gce_testing.go (hunk 3 above)
Comment (Subbarker) [reply integration_test/gce/gce_testing.go:1302]: Oops, that's wrong, fixed.

**Review by martijnvans — APPROVED** [review]

File: kokoro/scripts/test/go_test.sh (hunk 4 above)
Comment (Subbarker) [reply kokoro/scripts/test/go_test.sh]: Done

**Pull request merged**